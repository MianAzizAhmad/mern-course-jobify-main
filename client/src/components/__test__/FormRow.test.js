import React from 'react';
import ReactDom from 'react-dom';
import FormRow from './../FormRow';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

describe('Test Case { ', ()=>{

    it('FormRow Snap 1 }', ()=>{
        const tree = renderer.create(<FormRow type="text" name="name" value="test" handleChange="No change" labelText="testlabel"></FormRow>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('FormRow Snap 2 }', ()=>{
        const tree = renderer.create(<FormRow ></FormRow>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('FormRow Snap 3 }', ()=>{
        const tree = renderer.create(<FormRow type="password" name="snap3" value="SNAP" handleChange="Change" labelText="Password"></FormRow>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});