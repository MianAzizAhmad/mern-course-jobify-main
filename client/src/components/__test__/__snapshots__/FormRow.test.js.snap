// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case {  FormRow Snap 1 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
    htmlFor="name"
  >
    testlabel
  </label>
  <input
    className="form-input"
    name="name"
    onChange="No change"
    type="text"
    value="test"
  />
</div>
`;

exports[`Test Case {  FormRow Snap 2 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
  />
  <input
    className="form-input"
  />
</div>
`;

exports[`Test Case {  FormRow Snap 3 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
    htmlFor="snap3"
  >
    Password
  </label>
  <input
    className="form-input"
    name="snap3"
    onChange="Change"
    type="password"
    value="SNAP"
  />
</div>
`;
